File: bench/rb/fib_untyped.rb
# Untyped recursive Fibonacci - in Quoin this is the dispatch-heavy path (no
# annotations, full dynamic dispatch). In Ruby every call is dynamic anyway,
# so this is the same code as fib_typed.rb at the smaller workload (n=30),
# matching bench/qn/fib_untyped.qn.
# Run: `ruby bench/rb/fib_untyped.rb`.

class Fib
  def self.value(n)
    if n <= 1
      n
    else
      value(n - 1) + value(n - 2)
    end
  end
end

r = Fib.value(30)
if r == 832040
  puts 'fib_untyped: ok'
else
  puts "fib_untyped: FAIL got #{r}"
end

File: bench/py/fib_untyped.py
# Untyped recursive Fibonacci - Python port of bench/qn/fib_untyped.qn.
# Run: python3.13 bench/py/fib_untyped.py
#
# Kept as a class with a method (not a bare function) to preserve the
# dispatch-per-call shape of the Quoin original's `.value:` send.


class Fib:
    def value(self, n):
        if n <= 1:
            return n
        return self.value(n - 1) + self.value(n - 2)


r = Fib().value(30)
if r == 832040:
    print('fib_untyped: ok')
else:
    print('fib_untyped: FAIL got ' + str(r))
